storage/storage.go

```diff
@@ -154,10 +162,10 @@ func NewClient(ctx context.Context, opts ...option.ClientOption) (*Client, error
 
 		// Don't error out here. The user may have passed in their own HTTP
 		// client which does not auth with ADC or other common conventions.
-		c, err := transport.Creds(ctx, opts...)
+		c, err := internaloption.AuthCreds(ctx, opts)
```

Comment:
Does `internaloption.EnableNewAuthLibrary()` need to be added here (or elsewhere)?

Reply:
I don't think so, `internaloption.AuthCreds(ctx, opts)` will call into the new auth library no matter what it seems. What maybe should be added is a check for the env var to disable new auth if we still need that in case we need to debug?